### PR TITLE
[deckhouse] replace go-based conversions with jq-based

### LINKED ISSUE #8116: [deckhouse] add ModuleConfig conversions
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

We want to change ModuleConfig (config-values) schema and migrate users transparently between different versions

### Proposed Solution

1. Replace go-based conversion with file-based converions
2. Migrate old embedded conversions

https://docs.google.com/document/d/1QI2bJtv9TM6xeXvLt3-jUWywB9qEaWzsMcf1P4WZ_II/edit

### Additional Information

_No response_

### LINKED ISSUE #8189: [deckhouse] CI linter for module schema conversions
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

To avoid mistakes / misunderstandings

### Proposed Solution

Add the linter. Check, if module `openapi/` directory has conversions, `config-values.yaml` also must have `x-config-version: Y` field matching that conversions

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ipaqsa**
## Description
Provides replacement of go-based conversions with jq-based.

## Why do we need it, and what problem does it solve?
Closes #8116
Closes #8189 

## What is the expected result?
Embedded and external modules use jq-based conversion mechanism.

Conversions for extended-monitoring:
```
version: 2
conversions:
  - if .imageAvailability.skipRegistryCertVerification == true then .imageAvailability.registry.tlsConfig.insecureSkipVerify=true end
  - del(.imageAvailability.skipRegistryCertVerification) | if .imageAvailability == {} then del(.imageAvailability) end
```
```
root@dev-master-0:~# cat extended.yaml 
apiVersion: deckhouse.io/v1alpha1
kind: ModuleConfig
metadata:
  name: extended-monitoring
spec:
  version: 1
  enabled: true
  settings: # <-- Module parameters from the "Parameters" section below.
    imageAvailability:
      skipRegistryCertVerification: true
    events:
      exporterEnabled: true
      severityLevel: All
```
```
root@dev-master-0:~# k apply -f extended.yaml 
Warning: spec.version=1 is obsolete. Please migrate spec.settings to the latest version 2
moduleconfig.deckhouse.io/extended-monitoring created
```
```
root@dev-master-0:~# kubectl get moduleconfigs.deckhouse.io 
NAME                         ENABLED   VERSION   AGE     MESSAGE
extended-monitoring          true      1         24s     Update available, latest spec.settings schema version is 2
```
```
root@dev-master-0:~# dcm config extended-monitoring
events:
  exporterEnabled: true
  severityLevel: All
imageAvailability:
  registry:
    tlsConfig:
      insecureSkipVerify: true
```
```
root@dev-master-0:~# dcm values extended-monitoring
certificates:
  exporterEnabled: false
events:
  exporterEnabled: true
  severityLevel: All
imageAvailability:
  exporterEnabled: true
  registry:
    scheme: HTTPS
    tlsConfig:
      insecureSkipVerify: true
internal: {}

```

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: deckhouse
type: feature
summary: replace go-based conversions with jq-based
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ipaqsa. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ipaqsa. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 8783186282 8783186282-1-con-1-25 /sys/deckhouse-oss/install:pr8193 redos@158.160.107.103

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

File: deckhouse-controller/pkg/addon-operator/kube-config/backend/module_config.go [hunk 1]
```diff
@@ -173,25 +173,18 @@ func (mc ModuleConfigBackend) fetchValuesFromModuleConfig(item *v1alpha1.ModuleC
 		return utils.Values{}, nil
 	}
 
-	if item.Spec.Version == 0 {
```
Comment (yalosev) [inline deckhouse-controller/pkg/addon-operator/kube-config/backend/module_config.go:176]: I guess we can leave this optimization

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Review by miklezzzz — APPROVED** [review]

**Pull request merged**